src/Microsoft.Extensions.FileProviders.Physical/PollingFileChangeToken.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.Extensions.FileProviders.Physical
+{
+    public class PollingFileChangeToken : IChangeToken
+    {
+        private FileInfo _fileInfo;
+        private DateTime _previousWriteTimeUtc;
+        private DateTime _lastCheckedTimeUtc;
+
+        public PollingFileChangeToken(FileInfo fileInfo)
+        {
+            _fileInfo = fileInfo;
+            _previousWriteTimeUtc = GetLastWriteTimeUtc();
+        }
+
+        // Internal for unit testing
+        internal static TimeSpan PollingInterval { get; set; } = TimeSpan.FromSeconds(4);
+
+        private DateTime GetLastWriteTimeUtc()
+        {
+            _fileInfo.Refresh();
+            return _fileInfo.Exists ? _fileInfo.LastWriteTimeUtc : DateTime.MinValue;
+        }
+
+        public bool ActiveChangeCallbacks => false;
+
+        public bool HasChanged
```

Comment:
We might have to lock here to ensure that this behaves correctly when polled concurrently. A lot of these complexities show up because we have wildcard support in `Watch` and want to reuse instances of change token to avoid perform expensive regex matches. We could simplify a bunch of this here if we only ever support non-wildcard patterns.

cc @muratg \ @davidfowl thoughts?